src/sentry/api/endpoints/client_state.py

```diff
@@ -0,0 +1,55 @@
+# This endpoint helps managing persisted client state with a TTL for a member, organization or user
+
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.base import Endpoint
+from sentry.api.exceptions import ParameterValidationError
+from sentry.utils import json, redis
+
+STATE_CATEGORIES = {
+    "onboarding": {
+        "ttl": 30 * 24 * 60 * 60,  # the time in seconds that the state will be persisted
+        "scope": "member",  # Can be "user", "org", or "user-org"
+    }
+}
+
+
+class ClientStateEndpoint(Endpoint):
+    permission_classes = (IsAuthenticated,)
+
+    def get_key(self, request: Request, category):
```

Comment:
could you grab/validate the query parameter using a DRF serializer? we're trying to standardize using those as much as possible going forward for query params (can see example w/ `SCIMQueryParamSerializer`).

Reply:
Decided to use `OrganizationEndpoint`  so this is no longer needed now. Thanks for the tip tho